site/en/docs/android/custom-tabs/guide-interactivity/index.md

```diff
@@ -0,0 +1,197 @@
+authors:
+  - sebastianbenz
+---
+
+Now that we’ve customized the look and feel of your Custom Tabs, let’s focus in the second part of this tutorial on adding custom interactivity to Custom Tabs.
+
+## Enable the default share action 
+
+If you don't provide a custom share action, it is a good idea to enable the Share Action in the overflow menu, to enable users to share a link to the content they are seeing:
+
+```java
+    CustomTabsIntent.Builder intentBuilder = new CustomTabsIntent.Builder();
+    intentBuilder.setShareState(CustomTabsIntent.SHARE_STATE_ON);
+```
+
+## Add a custom action button 
+
+For a prominent action, the Custom Tab toolbar lets you integrate a custom action button, which can either have a custom icon. The icon should be 24dp in height and 24-48 dp in width. 
+
+{% Img src="image/6hHqS5auVgWhN0cQNQztaJx5w4M2/tYLvVrgICxyrpXlkzqkZ.png", alt="Custom Tab with a custom share action", width="400", height="866", class="screenshot screenshot--filled" %}
+
+For example, you can add a custom share action to the toolbar. To do this, create a [BroadcastReceiver](https://developer.android.com/guide/components/broadcasts#receiving-broadcasts) which gets called when the user clicks on the share action in the Custom Tab. 
+
+Register the BroadCastReceiver in the AndroidManifest.xml file:
+
+```xml
+<application …>
+  <receiver android:name=".ShareBroadcastReceiver" />
+</application>
+```
+
+Then add a new class `ShareBroadcastReceiver`. In the `onReceive` method, extract the currently displayed URL from the intent and trigger a send intent.
+
+```java
+public class ShareBroadcastReceiver extends BroadcastReceiver {
+   
+    @Override
+    public void onReceive(Context context, Intent intent) {
+        String url = intent.getDataString();
+        Intent sendIntent = new Intent();
+        sendIntent.setAction(Intent.ACTION_SEND);
+        sendIntent.putExtra(Intent.EXTRA_TEXT, url);
+        sendIntent.setType("text/plain");
+        Intent shareIntent = Intent.createChooser(sendIntent, null);
+        context.startActivity(shareIntent);
+    }
+}
+```
+
+Now, create a PendingIntent for ShareBroadcast and register it via [`setActionButton`](https://developer.android.com/reference/androidx/browser/customtabs/CustomTabsIntent.Builder#setActionButton(android.graphics.Bitmap,java.lang.String,android.app.PendingIntent)). Pass the pending intent together with the icon and the description. 
+
+```java
+String shareDescription = getString(R.string.label_action_share);
+Bitmap shareIcon = BitmapFactory.decodeResource(getResources(),
+       android.R.drawable.ic_menu_share);
+
+// Create a PendingIntent to your BroadCastReceiver implementation
+Intent actionIntent = new Intent(
+       this.getApplicationContext(), ShareBroadcastReceiver.class);
+PendingIntent pendingIntent =
+       PendingIntent.getBroadcast(getApplicationContext(), 0 /* request code */, actionIntent, PendingIntent.FLAG_MUTABLE);           
```

Comment:
what is the purpose of the comment? Does it have meaning to the expected reader?

Reply:
it should have meaning